### PR TITLE
Signin

### PR DESCRIPTION
**Pull request by wct9667**
# Description
Added presign in/up validation to the forms. On the sign-in, validation triggers on the submit getting hit, while on sign up validation is always triggered.

The confirm password doesn't pre-validate as I didn't want to save the password in a state on the page, but if it is fine, I can do that in combination with clearing the state to validate more than it being empty.

I initially used the built in validation, but made my own so I had more control over the styles and what was valid.

### CODE REVIEW

**Review by JustinBaez8 — CHANGES_REQUESTED** [review]
Would it be possible to only "validate" if the string input is not empty?

So once the user enters the sign up page, they are only shown the red outlines etc. once they start typing and not as soon as they arrive on the page?

Kinda like how the password seems to work at the moment:

image

Also, if possible, add this validation check to the update password form on the Dashboard, too.

File: src/components/forms/SignIn.tsx [hunk 1]
```diff
@@ -8,16 +8,40 @@ import logo from '../../../public/images/logos/Crowd_Comic_Logo_BW.svg';
...
     const togglePasswordVisibility = () => {
         setPasswordVisibility(!passwordVisible);
     };
 
+    const handleSubmit = (event: any) => {
+        if (!emailValid || !passwordValid) {
+            event.preventDefault();
+            event.stopPropagation();
+        }
+        setValidated(true);
```
Comment (nca6109) [inline src/components/forms/SignIn.tsx:31]: After submitting once, validated never gets set to false so each input field's error message will always update on change. Is this the intended function of the validation after a submit click?

Comment (wct9667) [reply src/components/forms/SignIn.tsx:31]: on the sign in page yes. Validated is meant to keep track if they hit submit the first time to then display the issues.

Comment (nca6109) [reply src/components/forms/SignIn.tsx:31]: Cool 👍

**Review by JustinBaez8 — COMMENTED** [review]
Password errors on the Dashboard are a bit jarring as they take up more space than necessary, making the whole div bigger:

image

We could also try the error toast on an "Could not find user with provided email/password" error on Sign In.

Also slight typo on invalid password on Sign In: "Password in invalid." Could maybe even remove the part saying that the password must be a certain length?

**Comment by nca6109** [comment]
Any reason that the validation errors come up under the input fields but this error is in a notification pop-up?
image
image

**Review by nca6109 — APPROVED** [review]
Besides the difference in front end and back end error notifications being a little jarring, everything looks great 😄

**Review by JustinBaez8 — APPROVED** [review]
Great job!

**Pull request merged**